main_test.go

```diff
@@ -544,6 +605,49 @@ func TestGetVMName(t *testing.T) {
 	}
 }
 
+func TestCheckIPv6Capable(t *testing.T) {
+	tests := []struct {
+		desc        string
+		httpHandler func(http.ResponseWriter, *http.Request)
+		wantOutput  bool
+	}{
+		{
+			desc: "case with check on ipv6 enabled host",
+			httpHandler: func(w http.ResponseWriter, r *http.Request) {
+				if r.Header.Get("Metadata-Flavor") != "Google" {
+					http.Error(w, "Missing Metadata-Flavor", 403)
+					return
+				}
+				w.Write([]byte("6970:7636:2061:6464:7265:7373:2062:6162"))
+			},
+			wantOutput: true,
+		},
+		{
+			desc: "case with check on non ipv6 host",
+			httpHandler: func(w http.ResponseWriter, r *http.Request) {
+				http.Error(w, "Not Found", 404)
+				return
+			},
+			wantOutput: false,
+		},
+	}
+	server := httptest.NewServer(nil)
+	defer server.Close()
+	overrideHTTP(server)
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			uuid.SetRand(rand.New(rand.NewSource(1)))
+			http.DefaultServeMux = new(http.ServeMux)
+			http.HandleFunc("http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/ipv6s", test.httpHandler)
+			want := true
```

Comment:
Shouldn't you be using `test.wantOutput` instead?

Reply:
yup thanks for catching that. when i fixed that it seems like there was a bug.. 
I was checking for err returned from `getFromMetadata`, but `getFromMetadata` does not return err for non-OK responses, and we return only the body from `getFromMetadata` -- so there is no way to check for status code from the caller. 
i'm adding addition logic in `getFromMetadata` to check for non-OK status codes. Lemeno what you think..